src/Gui/View3DInventorViewer.cpp

```diff
@@ -3341,6 +3342,49 @@ void View3DInventorViewer::viewSelection()
     }
 }
 
+void View3DInventorViewer::alignToSelection()
```

Comment:
Here high-level and low-level code is mixed. It would be good to refactor this function. E.g. the lines 3363-3381 can be moved to a new method in GeoFeature (or DocumentObject).

Reply:
I have changed some things here, is it better now?

Reply:
Yes, it's much better now.